R/format-report.R

```diff
@@ -683,18 +694,40 @@ format_traceability_matrix <- function(
       )
 
     # Get testing directories for caption
-    test_dirs <- exported_func_df %>% tidyr::unnest(test_dirs) %>% dplyr::pull(test_dirs) %>% unique()
-    test_dirs <- test_dirs[test_dirs != ""] %>% paste(collapse = ", ")
+    if ("test_dirs" %in% names(exported_func_df)) {
+      test_dirs <- exported_func_df %>% tidyr::unnest(test_dirs) %>% dplyr::pull(test_dirs) %>% unique()
+      test_dirs <- test_dirs[test_dirs != ""] %>% paste(collapse = ", ")
+      # Remove testing directory column (not a column due to horizontal space limits)
+      exported_func_df <- exported_func_df %>% dplyr::select(-"test_dirs")
+    } else {
+      test_dirs <- NULL
+    }
+
+    if ("exported_function" %in% names(exported_func_df)) {
+      # Align internal scoring with external format.
+      exported_func_df <- dplyr::rename(exported_func_df,
+        entrypoint = "exported_function"
+      )
+    }
 
-    # Remove testing directory column (not a column due to horizontal space limits)
-    exported_func_df <- exported_func_df %>% dplyr::select(-"test_dirs")
+    entry_name <- switch(scorecard_type,
+      "R" = "Exported Function",
+      "cli" = "Command",
+      "Entry Point"
+    )
+    exported_func_df <- dplyr::rename(
+      exported_func_df,
+      !!entry_name := "entrypoint"
+    )
 
     # Format Table
     if(isTRUE(wrap_cols)){
       exported_func_df <- exported_func_df %>%
         dplyr::mutate(
-          dplyr::across("exported_function":"documentation", ~
-                          wrap_text(.x, width = 24, indent = TRUE, strict = TRUE)),
+          dplyr::across(
+            all_of(c(entry_name, "code_file", "documentation")),
+            function(x) wrap_text(x, width = 24, indent = TRUE, strict = TRUE)
```

Comment:
I don't think this is _quite_ the right section of the code for this comment, but it seems close enough...

I noticed that, in the Traceability Matrix, the text-wrapping on the Documentation column looks nicer than in the Test Files column. Specifically, it attempts to split on separators and indents the wrapped lines. For example:

![Screenshot 2024-08-09 at 5 12 37 PM](https://github.com/user-attachments/assets/548d8217-a1c3-49dd-aa0c-c6ffcd7bd69d)

I'm assuming there's a reason for this, but if it's easy enough to fix, can we make the Test Files column have the nicer wrapping too?

This is a bit of a nit pick though, so if there's an implementation reason why this is difficult, it's really not necessary to spend much time on it.

Reply:
> I'm assuming there's a reason for this, but if it's easy enough to fix, can we make the Test Files column have the nicer wrapping too?

I don't know what the original rationale was (hopefully @barrettk can chime in), but, when working on gh-69, I recall wondering a bit about it and then realizing that indenting, as currently implemented by `wrap_text`, shouldn't be applied because that would lead to confusing indentation for the remaining files.  (Basically, I think it's meant only for scalar values.)

Example with `indent = TRUE` and default `wrap_sym` characters:


```diff
diff --git a/R/format-report.R b/R/format-report.R
index 1b24d0a..7b248fb 100644
--- a/R/format-report.R
+++ b/R/format-report.R
@@ -730,7 +730,7 @@ format_traceability_matrix <- function(
           ),
           # Tests can be longer due to page width (pg_width) settings (we make it wider)
           test_files = purrr::map_chr(.data$test_files, function(tests){
-            wrap_text(tests, width = 40, strict = TRUE, wrap_sym = NULL)
+            wrap_text(tests, width = 40, indent = TRUE, strict = TRUE)
           })
         )
     }
```

![20240809T182006](https://github.com/user-attachments/assets/1b2ed647-90f2-4cf8-baad-a8d60727ea7e)

Still, I think it'd be an improvement to at least wrap on the default `wrap_sym` characters (at least in this example):

```diff
diff --git a/R/format-report.R b/R/format-report.R
index 1b24d0a..e6d120f 100644
--- a/R/format-report.R
+++ b/R/format-report.R
@@ -730,7 +730,7 @@ format_traceability_matrix <- function(
           ),
           # Tests can be longer due to page width (pg_width) settings (we make it wider)
           test_files = purrr::map_chr(.data$test_files, function(tests){
-            wrap_text(tests, width = 40, strict = TRUE, wrap_sym = NULL)
+            wrap_text(tests, width = 40, strict = TRUE)
           })
         )
     }
```

![20240809T182207](https://github.com/user-attachments/assets/42b0cc86-e46a-45db-bf5b-15d5f4be044f)

Reply:
I looked over the PR discussion and commit messages in the original `wrap_sym` PR (gh-57), and I didn't spot any discussion of why `test_files` uses `wrap_sym = NULL`.

I'll add a commit here that drops `wrap_sym = NULL`.

Reply:
I believe `wrap_sym = NULL` was set for test files to avoid the issue you highlighted above (caused too many new lines that made it harder to read). Indents were not meant to be made for additional elements in a vector (i.e. multiple test files), but rather only when a newline was introduced to due to the column width. Could be worth adjusting in the future but I think your second screenshot looks good